File: Data Structures & Algorithms/anagram-groups/submission-3.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let res = new Map()
        for(const s of strs){
            const sortedS = s.split('').sort().join('');

            if(!res.has(sortedS)){
                res.set(sortedS, [])
            }

            res.get(sortedS).push(s)

        }
        return Array.from(res.values());

    }
    
}


/**
 * 
 res = collections.defaultdict(list)
 for s in strs:
    res[tuple(Counter(s))].append(s)
return list(res.values())
 */

File: Data Structures & Algorithms/longest-common-prefix/submission-3.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        n = min(len(s) for s in strs)
        target = strs[0]
        res = 0
        i = 0
        while i < n and all(s[i] == target[i] for s in strs):
            res += 1
            i += 1
        return target[:res]
